esp-hal-common/src/dma/gdma.rs

```diff
@@ -219,9 +361,11 @@ macro_rules! ImplChannel {
 pub(crate) mod private {
     use crate::dma::{private::*, *};
 
-    ImplChannel!(0);
-    ImplChannel!(1);
-    ImplChannel!(2);
+    impl_channel!(0);
+    #[cfg(not(esp32c2))]
+    impl_channel!(1);
+    #[cfg(not(esp32c2))]
+    impl_channel!(2);
```

Comment:
for ESP32-S3 we can add two more channels

Reply:
Good catch, thanks! I'll add those in when I fix the priority.